templates/reputation/test/reputation.js

```diff
@@ -0,0 +1,139 @@
+      const resolvedAddress = await PublicResolver.at(await ens.resolver(aragonIdNameHash)).addr(aragonIdNameHash)
+      assert.equal(resolvedAddress, dao.address, 'aragonId ENS name does not match')
+    })
+
+    it('should have voting app correctly setup', async () => {
+      assert.isTrue(await voting.hasInitialized(), 'voting not initialized')
+      assert.equal((await voting.supportRequiredPct()).toString(), 50e16)
+      assert.equal((await voting.minAcceptQuorumPct()).toString(), 20e16)
+      assert.equal((await voting.voteTime()).toString(), 60 * 60 * 24 * 7)
+
+      await assertRole(acl, voting, voting, 'CREATE_VOTES_ROLE', tokenManager)
+      await assertRole(acl, voting, voting, 'MODIFY_QUORUM_ROLE')
+      await assertRole(acl, voting, voting, 'MODIFY_SUPPORT_ROLE')
+    })
+
+    it('should have token correctly setup', async () => {
+      assert.equal(await token.name(), 'Reputation Token')
+      assert.equal(await token.symbol(), 'REP')
+      assert.equal((await token.decimals()).toString(), 18)
+      assert.equal((await token.totalSupply()).toString(), STAKES.reduce((a, b) => a + b))
+
+      for (const holder of HOLDERS) assert.equal((await token.balanceOf(holder)).toString(), STAKES[HOLDERS.indexOf(holder)])
+    })
+
+    it('should have token manager app correctly setup', async () => {
+      assert.isTrue(await tokenManager.hasInitialized(), 'token manager not initialized')
+      assert.equal(await tokenManager.token(), token.address)
+
+      await assertRole(acl, tokenManager, voting, 'MINT_ROLE')
+      await assertRole(acl, tokenManager, voting, 'BURN_ROLE')
+    })
+
+    it('should have finance app correctly setup', async () => {
+      assert.isTrue(await finance.hasInitialized(), 'finance not initialized')
+      assert.equal(web3.toChecksumAddress(await finance.vault()), agent.address)
+
+      await assertRole(acl, finance, voting, 'CREATE_PAYMENTS_ROLE')
+      await assertRole(acl, finance, voting, 'EXECUTE_PAYMENTS_ROLE')
+      await assertRole(acl, finance, voting, 'MANAGE_PAYMENTS_ROLE')
+    })
+
+    it('should have agent app correctly setup', async () => {
+      assert.isTrue(await agent.hasInitialized(), 'agent not initialized')
+      assert.equal(await agent.designatedSigner(), ZERO_ADDRESS)
+
+      await assertRole(acl, agent, voting, 'EXECUTE_ROLE')
+      await assertRole(acl, agent, voting, 'RUN_SCRIPT_ROLE')
+      await assertRole(acl, agent, voting, 'TRANSFER_ROLE', finance)
+    })
+
+    it('setup DAO and ACL permissions correctly', async () => {
+      await assertRole(acl, dao, voting, 'APP_MANAGER_ROLE')
+      await assertRole(acl, acl, voting, 'CREATE_PERMISSIONS_ROLE')
+    })
+
+    it('setup EVM scripts registry permissions correctly', async () => {
+      const reg = await EVMScriptRegistry.at(await acl.getEVMScriptRegistry())
+      await assertRole(acl, reg, voting, 'REGISTRY_ADD_EXECUTOR_ROLE')
+      await assertRole(acl, reg, voting, 'REGISTRY_MANAGER_ROLE')
```

Comment:
A thought that just came up... would it also make sense to check that the other roles don't have permission managers for all these apps here and in the other templates (we can't check for grantees, since we can't check all addresses)?